.github/workflows/build_and_test_cosmian_vm.yml

```diff
@@ -0,0 +1,174 @@
+                                                            --tags ssh-full,backend,http-server,https-server,cosmian-vm-agent \
+                                                            --metadata-from-file=startup-script=scripts/gcp-start.sh \
+                                                            --max-run-duration=10m \
+                                                            --instance-termination-action=DELETE
+          IP_ADDR=$(gcloud beta compute instances describe $CI_INSTANCE --format='get(networkInterfaces[0].accessConfigs[0].natIP)' --zone=europe-west4-a)
+          echo "IP_ADDR=${IP_ADDR}" >> "$GITHUB_OUTPUT"
+
+      - name: Download Cosmian VM client
+        run: |
+          wget https://package.cosmian.com/cosmian_vm/1.0.1/cosmian_vm
+
+      - name: Change permissions of binaries
+        run: |
+          chmod +x ./cosmian_vm
+
+      - name: Test Cosmian VM Agent on GCP instance
+        env:
+          IP_ADDR: ${{ steps.run-gcp-instance.outputs.IP_ADDR }}
+        run: |
+          echo "Waiting for Cosmian VM agent..."
+          until curl --insecure --output /dev/null --silent --fail https://${IP_ADDR}:5355/ima/ascii; do sleep 3; done
+          echo "[ OK ] Cosmian VM ready"
+          ./cosmian_vm --url https://${IP_ADDR}:5355 --allow-insecure-tls snapshot
+          ./cosmian_vm --url https://${IP_ADDR}:5355 --allow-insecure-tls verify --snapshot ./cosmian_vm.snapshot
+
+      - name: Test KMS on the Cosmian VM
+        env:
+          IP_ADDR: ${{ steps.run-gcp-instance.outputs.IP_ADDR }}
+        run: |
+          echo "Checking Cosmian KMS HTTP connection..."
+          curl http://${IP_ADDR}:8080/version
+          echo ""
+          echo "[ OK ] Cosmian KMS HTTP connection"
+          echo "Checking Cosmian KMS HTTPS connection..."
+          curl --insecure https://${IP_ADDR}/version
+          echo ""
+          echo "[ OK ] Cosmian KMS HTTPS connection"
+          echo "Checking Cosmian KMS HTTP to HTTPS redirect connection..."
+          curl --insecure http://${IP_ADDR}/version
+          echo ""
+          echo "[ OK ] Cosmian KMS HTTP to HTTPS redirect connection"
+
+      - name: Stop and delete GCP instance
+        if: success() || failure()
+        env:
+          CI_INSTANCE: ${{ needs.build-kms-cosmian-vm-image.outputs.ci_instance }}
+        run: |
+          set +e
+          OUTPUT=$(gcloud beta compute instances describe $CI_INSTANCE --format='get(status)' --zone=europe-west4-a)
+          EXIT_CODE="$?"
+          if [ "$EXIT_CODE" -eq 0 ]; then
+            gcloud beta compute instances stop $CI_INSTANCE --zone europe-west4-a \
+                                                            --project $GCP_DEV_PROJECT
+            gcloud beta compute instances delete $CI_INSTANCE --zone europe-west4-a \
+                                                              --project $GCP_DEV_PROJECT
+          fi
+
+  cleanup:
```

Comment:
in my opinion, we could let the nightly build do this cleanup.

Reply:
no strong opinion about this, let's remove this